test/e2e/service.go

```diff
@@ -1010,66 +1011,63 @@ func testLoadBalancerNotReachable(ingress api.LoadBalancerIngress, port int) {
-	for t := time.Now(); time.Since(t) < podStartTimeout; time.Sleep(5 * time.Second) {
-		resp, err = httpGetNoConnectionPool(url)
-		if err == nil {
-			break
+	expectNoError(wait.Poll(poll, podStartTimeout, func() (bool, error) {
+		resp, err := httpGetNoConnectionPool(url)
+		if err != nil {
+			Logf("Got error waiting for reachability of %s: %v", url, err)
+			return false, nil
 		}
-		By(fmt.Sprintf("Got error waiting for reachability of %s: %v", url, err))
-	}
-	Expect(err).NotTo(HaveOccurred())
-	defer resp.Body.Close()
-
-	body, err := ioutil.ReadAll(resp.Body)
-	Expect(err).NotTo(HaveOccurred())
-	if resp.StatusCode != 200 {
-		Failf("received non-success return status %q trying to access %s; got body: %s", resp.Status, url, string(body))
-	}
-	if !strings.Contains(string(body), "test-webserver") {
-		Failf("received response body without expected substring 'test-webserver': %s", string(body))
-	}
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			Logf("Got error reading response from %s: %v", url, err)
+			return false, nil
+		}
+		if resp.StatusCode != 200 {
+			return false, fmt.Errorf("received non-success return status %q trying to access %s; got body: %s", resp.Status, url, string(body))
+		}
+		if !strings.Contains(string(body), "test-webserver") {
+			return false, fmt.Errorf("received response body without expected substring 'test-webserver': %s", string(body))
+		}
+		Logf("Successfully reached %v", url)
+		return true, nil
+	}))
 }
 
 func testNotReachable(ip string, port int) {
-	var err error
-	var resp *http.Response
-	var body []byte
-
 	url := fmt.Sprintf("http://%s:%d", ip, port)
 	if ip == "" {
-		Failf("got empty IP for non-reachability check", url)
+		Failf("Got empty IP for non-reachability check", url)
 	}
 	if port == 0 {
-		Failf("got port==0 for non-reachability check", url)
+		Failf("Got port==0 for non-reachability check", url)
 	}
 
-	for t := time.Now(); time.Since(t) < podStartTimeout; time.Sleep(5 * time.Second) {
-		resp, err = httpGetNoConnectionPool(url)
+	By(fmt.Sprintf("Checking that %s is not reachable", url))
+	expectNoError(wait.Poll(poll, podStartTimeout, func() (bool, error) {
+		resp, err := httpGetNoConnectionPool(url)
 		if err != nil {
-			break
+			Logf("Expecting %s to be unreachable but is still reachable", url)
+			return true, nil
 		}
-		body, err = ioutil.ReadAll(resp.Body)
-		Expect(err).NotTo(HaveOccurred())
-		resp.Body.Close()
-		By(fmt.Sprintf("Got success waiting for non-reachability of %s: %v", url, resp.Status))
-	}
-	if err == nil {
-		Failf("able to reach service %s when should no longer have been reachable: %q body=%s", url, resp.Status, string(body))
-	}
-	// TODO: Check type of error
-	By(fmt.Sprintf("Found (expected) error during not-reachability test %v", err))
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			Logf("Got error reading response from %s: %v", url, err)
```

Comment:
OK this is going to seem super nit-y, but I think it's important to have "success" or "successfully" in a string when describing an expected failed something. I think if people see "The url X is unreachable", it looks like an error message. If it says "Successfully found X to be unreachable." then it's ultra-clear.

We've had some code internally that used this affirmatively-negative logging and it's been really nice when we've had to pour through its output when it's failing.